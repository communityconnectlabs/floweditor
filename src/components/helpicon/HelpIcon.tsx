import React from 'react';
import ReactTooltip from 'react-tooltip';
import styles from './HelpIcon.module.scss';

export interface HelpIconProps {
  iconColor: string;
  iconSize: string;
  dataFor: string;
  children: JSX.Element[] | JSX.Element;
}

const HelpIcon = (props: HelpIconProps): JSX.Element => (
  <>
    <div
      className={styles.icon}
      style={{ color: props.iconColor, fontSize: props.iconSize }}
      data-tip=""
      data-for={props.dataFor}
    >
      <span className="fe-help" />
    </div>
    <ReactTooltip id={props.dataFor} effect="solid" multiline={true} className={styles.tooltip}>
      {props.children}
    </ReactTooltip>
  </>
);

export default HelpIcon;
