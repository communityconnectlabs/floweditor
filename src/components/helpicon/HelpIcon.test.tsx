import React from 'react';
import { composeComponentTestUtils } from 'testUtils';
import HelpIcon, { HelpIconProps } from './HelpIcon';

const baseProps: HelpIconProps = {
  iconColor: '#000',
  iconSize: '15px',
  dataFor: 'testNode',
  children: <p>Some test tooltip text</p>
};

const { setup } = composeComponentTestUtils(HelpIcon, baseProps);

describe('HelpIcon component', () => {
  it('should render helper icon with tooltip', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
