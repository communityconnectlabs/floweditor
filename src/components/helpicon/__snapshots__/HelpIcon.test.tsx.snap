// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpIcon component should render helper icon with tooltip 1`] = `
<Fragment>
  <div
    className="icon"
    data-for="testNode"
    data-tip=""
    style={
      Object {
        "color": "#000",
        "fontSize": "15px",
      }
    }
  >
    <span
      className="fe-help"
    />
  </div>
  <ReactTooltip
    className="tooltip"
    clickable={false}
    delayHide={300}
    effect="solid"
    id="testNode"
    insecure={true}
    multiline={true}
    resizeHide={true}
    wrapper="div"
  />
</Fragment>
`;
